### PR TITLE
kiwmi_output:usable_area() and :on("usable_area_change")

### PR DESCRIPTION
**Pull request by tiosgz**
This is especially useful for window tiling and maximizing.

(another PR from `tiosgz/mine`, commits f378b29a61fc960711ab69b54d1d7d3bbc76a782 and b0569ea72bb5482dd760c66eae866d2b8844a621)

### CODE REVIEW

File: kiwmi/desktop/layer_shell.c [hunk 1]
```diff
@@ -301,6 +301,12 @@ arrange_layers(struct kiwmi_output *output)
         }
     }
 
+    if (memcmp(&usable_area, &output->usable_area,
+                sizeof(struct wlr_box)) != 0) {
```
Comment (buffet) [inline kiwmi/desktop/layer_shell.c:305]: Prefer sizeof(usable_area).

**Review by buffet — CHANGES_REQUESTED** [review]
Thanks! I was actually just discussing this with a friend!

Could you run clang-format (`clang-format -i **/*.{c,h}`) over the project please?

File: kiwmi/desktop/layer_shell.c (hunk 1 above)
Comment (tiosgz) [reply kiwmi/desktop/layer_shell.c]: Sure; done.

**Review by buffet — APPROVED** [review]
Thanks!

**Pull request merged**